sql/sql_repl.cc

```diff
@@ -1870,7 +1883,8 @@ send_event_to_slave(binlog_send_info *info, Log_event_type event_type,
           info->gtid_skip_group = (flags2 & Gtid_log_event::FL_STANDALONE ?
                               GTID_SKIP_STANDALONE : GTID_SKIP_TRANSACTION);
         }
-        else if (event_gtid.server_id == gtid->server_id &&
+        else if (gtid != NULL &&
+                 event_gtid.server_id == gtid->server_id &&
```

Comment:
... so that you avoid a double check for NULL here.

Eg.
  if (gtid==NULL) {
    if (!info->is_until_before_gtids) {
    ...
    }
} else if (event_gtid.server_id == gtid->server_id &&
                 event_gtid.seq_no >= gtid->seq_no) {
  ...

Reply:
Oh nice! I like that :)